### PR TITLE
ENH: Better error message for invalid axis

### PR DESCRIPTION
**Pull request by shoyer**
xref #7915

Currently, writing array.max(axis=[0, 1]) doesn't mention you can use a tuple
for the axis argument -- it just says "an integer is required".

### CODE REVIEW

File: numpy/core/src/multiarray/conversion_utils.c [hunk 1]
```diff
@@ -289,7 +289,7 @@ PyArray_ConvertMultiAxis(PyObject *axis_in, int ndim, npy_bool *out_axis_flags)
         memset(out_axis_flags, 0, ndim);
 
         axis = PyArray_PyIntAsInt_ErrMsg(axis_in,
-                                   "an integer is required for the axis");
+                               "an integer or tuple is required for the axis");
```
Comment (charris) [inline numpy/core/src/multiarray/conversion_utils.c]: Maybe `tuple of integers`?

File: numpy/core/tests/test_ufunc.py [hunk 2]
```diff
@@ -1197,6 +1197,7 @@ def test_reduce_arguments(self):
         assert_raises(TypeError, f, d, axis="invalid")
         assert_raises(TypeError, f, d, axis="invalid", dtype=None,
                       keepdims=True)
+        assert_raises_regex(TypeError, "integer or tuple", f, d, axis=[0])
```
Comment (charris) [inline numpy/core/tests/test_ufunc.py]: Apparently a list raises a different error.

Comment (charris) [inline numpy/core/tests/test_ufunc.py]: Seems to come from python rather than numpy.

Comment (juliantaylor) [inline numpy/core/tests/test_ufunc.py]: it comes from the PyNumber_Index here https://github.com/numpy/numpy/blob/master/numpy/core/src/multiarray/conversion_utils.c#L848

**Comment by charris** [comment]
@shoyer Want to keep working on this?

**Comment by shoyer** [comment]
I don't know what's going on here with the including the appropriate functions -- not sure when/if I'll have time to debug this.

**Comment by charris** [comment]
Mind if I close this then? You can reopen it if/when you have the time.

**Comment by shoyer** [comment]
Sure

**Comment by charris** [comment]
OK, closing. Reopen when ready.

**Pull request closed without merging**